src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -69,15 +73,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     lectureEntity.setProperty(Lecture.PROP_URL, videoUrl);
     lectureEntity.setProperty(Lecture.PROP_ID, videoId);
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(lectureEntity);
-    response.sendRedirect(REDIRECT_URL);
+    response.sendRedirect(
+        buildRedirectUrl(String.valueOf(lectureEntity.getKey().getId()), videoId));
```

Comment:
Maybe have buildRedirectUrl take an entity or Key instead of string? Because I assume you'll never be passing something other than an lecture's key